cmd/kube-scheduler/app/options/options.go

```diff
@@ -237,15 +238,19 @@ func (o *Options) Config() (*schedulerappconfig.Config, error) {
 	}
 
 	// Prepare event clients.
-	eventBroadcaster := events.NewBroadcaster(&events.EventSinkImpl{Interface: eventClient.EventsV1beta1().Events("")})
-	recorder := eventBroadcaster.NewRecorder(scheme.Scheme, c.ComponentConfig.SchedulerName)
-	leaderElectionBroadcaster := record.NewBroadcaster()
-	leaderElectionRecorder := leaderElectionBroadcaster.NewRecorder(scheme.Scheme, corev1.EventSource{Component: c.ComponentConfig.SchedulerName})
+	if _, err := client.Discovery().ServerResourcesForGroupVersion(eventsv1beta1.SchemeGroupVersion.String()); err == nil {
+		c.Broadcaster = events.NewBroadcaster(&events.EventSinkImpl{Interface: eventClient.EventsV1beta1().Events("")})
+		c.Recorder = c.Broadcaster.NewRecorder(scheme.Scheme, c.ComponentConfig.SchedulerName)
+		c.EventClient = eventClient.EventsV1beta1()
```

Comment:
do all call sites tolerate these three fields being nil?

Reply:
not yet, but idea is to do so